sqlalchemy_utils/functions/database.py

```diff
@@ -459,41 +462,49 @@ def sqlite_file_exists(database):
 
     url = copy(make_url(url))
     database, url.database = url.database, None
-    engine = sa.create_engine(url)
+    engine = None
 
-    if engine.dialect.name == 'postgresql':
-        text = "SELECT 1 FROM pg_database WHERE datname='%s'" % database
-        return bool(get_scalar_result(engine, text))
+    if engine.dialect.name.startswith('postgresql'):
```

Comment:
Shouldn't this raise an `AttributeError` since `engine` is set to `None`?

Reply:
true

Reply:
would it be an alternatve to use `url.drivername` instead. Otherwise I would revert to creating the engine before the if-else.

Reply:
I wonder why the tests didn't fail?

I think you can set `dialect_name = url.get_dialect().name` before the `if` and then
```suggestion
    if dialect_name == 'postgresql':
```
and similarly for the following `elif`